ia-wishlist-bot/adding_wishlist_ol.py

```diff
@@ -0,0 +1,132 @@
+        expected = {"title": "Larks in a paradise: New Zealand portraits", "authors": [
+            'McNeish, James', 'Friedlander, Marti'], "language": "eng", "date": "1974", "oclc": "16289249", "isbn10": "0002114976", "isbn13": "9780002114974"}
+
+        self.assertTrue(parse_wishlist_csv_row_to_dict(csv) == expected)
+
+    def test_get_author_object(self):
+        author = {'author_name': 'JK Rowling'}
+        expected = common.Author(name='JK Rowling')
+        author_obj = get_author_object(author.get('author_name'), author.get('author_birth_date'), author.get('author_death_date'))
+
+        self.assertTrue(expected.name == author_obj.name and expected.identifiers == author_obj.identifiers)
+
+    def test_get_bookcover(self):
+        csv = ["Mom Goes to War(Light)", "[' Irene Aparici Martin']","eng", "1974", "16289249", "8415503202", "9788415503200"]
+
+        book = parse_wishlist_csv_row_to_dict(csv)
+        expected_url = "https://images.betterworldbooks.com/841/9788415503200.jpg"
+
+        self.assertTrue(expected_url == get_bookcover(book))
+
+    def test_empty_bookcover(self):
+        csv = ["Alicia a trave s del espejo La caza del snark", "['Lewis Carroll']",
+               "Spanish", "2002", "893562252", "9706664998", "9789706664990"]
+        book = parse_wishlist_csv_row_to_dict(csv)
+        expected_url = None
+
+        self.assertTrue(expected_url == get_bookcover(book))
+
+
+def process_csv(filename):
+    """This function takes a csv file which was output from our whatever process created e.g. *new_wishlist_salman_1000.csv* and converts it into a python dictionary
+
+    Usage:
+        >>> parse_wishlist_csv_row_to_dict("foo,bar,baz,qux")
+        { "author": "foo", "title": "bar", ...}
+    """
+    with open(filename, mode='r') as infile:
+        reader = csv.reader(infile)
+
+        book_data = [row for row in reader]
+
+        return book_data
+
+def parse_wishlist_csv_row_to_dict(csv):
+    """This function takes a csv row which was output from our whatever process created e.g. *new_wishlist_salman_1000.csv* and converts it into a python dictionary
+
+    Usage:
+        >>> parse_wishlist_csv_row_to_dict("foo,bar,baz,qux")
+        { "author": "foo", "title": "bar", ...}
+    """
+    book = {"title": csv[0], "authors": ast.literal_eval(csv[1]), "language": csv[2], "date": csv[3], "oclc": csv[4], "isbn10": csv[5], "isbn13": csv[6]}
+    return book
+
+
+def get_author_object(author_name, author_birth_date=None, author_death_date=None):
+    """This takes an author name which was output from our CSV row and then either retrieves an ol Object which was already created or creates a new OL Object
+
+    Usage:
+        >>> get_author_object('Dan Brown')
+    """
+    author_olid = ol.Author.get_olid_by_name(author_name)
+    if author_olid:
+        return ol.get(author_olid)
+    else:
+        return common.Author(name=author_name)
```

Comment:
This will produce an author like this:
https://openlibrary.org/authors/OL7416671A/Eagle_Audrey_Lily on
https://openlibrary.org/books/OL26468985M/Eagle's_Trees_and_shrubs_of_New_Zealand.

When there is actually an existing author of: https://openlibrary.org/authors/OL1784176A/Audrey_Lily_Eagle

When we parse the author name we want to make sure there is not a ","  -- it shouldn't be Last name, First name.

Reply:
we may want to add a very small regression test just to make sure within openlibrary-client that ',' is not allowed in an author name.

Reply:
cc: @hornc